Add `wrangler delete` for scripts

ocsfrank opened the pull request:
This PR adds a command `wrangler delete` that allows to delete a Workers script.

`wrangler delete` will delete a script based on the script name and account ID found in the `wrangler.toml` file. 

Without any flags, the command will ask for confirmation steps when deleting the script, and any bound Durable Objects.
It's also possible to pass a flag `--force` to skip all the confirmation steps. **WARNING:** `--force` **will also delete the Durable Objects bound to a script.** 

Should solve #210.

Electroid commented on src/cli/delete.rs:
```diff
@@ -0,0 +1,20 @@
+        if let (Some(account_id), Some(script_id)) = (account_id, script_id) {
+            commands::delete::delete_script(&user, force, &account_id, &script_id)
+        } else {
+            anyhow::bail!("Both account-id and script-id must be provided.")
```
I think `script-name` would be more consistent with how we reference this elsewhere.

Electroid commented on src/cli/mod.rs:
```diff
@@ -201,6 +203,39 @@ pub enum Command {
+    #[structopt(name = "delete")]
+    Delete {
+        /// Disables interactive mode
+        #[structopt(long = "no-interactive", short = "n")]
```
`non-interactive`

Electroid commented on src/cli/mod.rs:
```diff
@@ -201,6 +203,39 @@ pub enum Command {
+            requires = "no-interactive",
+            requires = "script-id"
+        )]
+        account_id: Option<String>,
```
Even in non-interactive mode, we we use the default `account_id`? (from the credentials) If there are more than 1 options, then we can require it.

jyn514 replied:
Hmm, wrangler uses the account_id from `wrangler.toml` everywhere else, right? Why have it as a command line parameter here instead?

ocsfrank replied:
The idea I had in mind was to let users run `wrangler delete` from any directory even if they don't have a Workers script on their machine.

nilslice replied:
> The idea I had in mind was to let users run wrangler delete from any directory even if they don't have a wrangler.toml.

I can see this being a nice feature, but since we don't support deploying a project w/o a `wrangler.toml`, then it's probably safer & more consistent to maintain the same restriction when deleting. 

It is not uncommon for other deployment systems to _reference_ code artifacts to deploy (or tear down) without having the artifact / source locally though. So I do see this as generally a good idea, but maybe not for now for wrangler as it stands.

ocsfrank replied:
> > The idea I had in mind was to let users run wrangler delete from any directory even if they don't have a wrangler.toml.
> 
> I can see this being a nice feature, but since we don't support deploying a project w/o a `wrangler.toml`, then it's probably safer & more consistent to maintain the same restriction when deleting.
> 
> It is not uncommon for other deployment systems to _reference_ code artifacts to deploy (or tear down) without having the artifact / source locally though. So I do see this as generally a good idea, but maybe not for now for wrangler as it stands.

I see, that makes a lot of sense. Thanks everyone for the feedback!

Electroid commented on src/commands/delete/mod.rs:
```diff
@@ -0,0 +1,201 @@
+    let accounts = fetch_accounts(user)?;
+    let (valid_accounts, table) = format_accounts(accounts)?;
+    println!("{}", &table);
+    let account_name = interactive::get_user_input(
+        "Please enter the name of the account you wish to delete a Workers script from",
+    );
```
Can this be skipped if they have 1 account?

jyn514 commented on src/commands/delete/mod.rs:
```diff
@@ -0,0 +1,201 @@
+    let accounts = fetch_accounts(user)?;
+    let (valid_accounts, table) = format_accounts(accounts)?;
+    println!("{}", &table);
+    let account_name = interactive::get_user_input(
+        "Please enter the name of the account you wish to delete a Workers script from",
+    );
+    if !valid_accounts.contains_key(&account_name) {
+        anyhow::bail!("Account name doesn't match.")
+    }
+
+    //  Fetch and display scripts related to the account
+    let account_id = valid_accounts.get(&account_name).unwrap();
```
This seems really different from how IDs are handled everywhere else. This should use `manifest.account_id.load()` instead.

ocsfrank commented:
Changed to read all project info from `wrangler.toml` and added a PR to `cloudflare-rs` (https://github.com/cloudflare/cloudflare-rs/pull/166) for adding the API endpoints.

JacobMGEvans commented:
Thanks for creating this PR! Wrangler v1 is now deprecated and support is only being provided for critical updates or security concerns. As such, we are closing this PR.

New versions of Wrangler are maintained in the [workers-sdk repo](https://github.com/cloudflare/workers-sdk). If you are running into a similar issue with wrangler v2, please report it or create a new PR in the [workers-sdk repo](https://github.com/cloudflare/workers-sdk). For more info about wrangler v1 deprecation, please check out our [blog post](https://blog.cloudflare.com/deprecating-wrangler-v1).
